src/Cardano/HttpBridge/Client.hs

```diff
@@ -0,0 +1,51 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE RankNTypes #-}
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+-- | An API client for the Cardano HTTP Bridge.
+module Cardano.HttpBridge.Client
+    ( getBlockByHash
+    , getEpochById
+    , getTipBlockHeader
+    ) where
+
+import Cardano.HttpBridge.Api
+    ( api )
+import Cardano.Network.Name
+    ( NetworkName (getNetworkName) )
+import Cardano.Wallet.Binary
+    ()
+import Cardano.Wallet.Primitive
+    ( Block, BlockHeader, EpochId (getEpochId), Hash, getHash )
+import Data.Text.Encoding
+    ( decodeUtf8 )
+import Prelude
+import Servant.API
+    ( (:<|>) (..), ToHttpApiData (..) )
+import Servant.Client
+    ( ClientM, client )
+
+-- | Retrieve a block identified by the unique hash of its header.
+getBlockByHash :: NetworkName -> Hash "BlockHeader" -> ClientM Block
+
+-- | Retrieve all the blocks for the epoch identified by the given integer ID.
+getEpochById :: NetworkName -> EpochId -> ClientM [Block]
+
+-- | Retrieve the header of the latest known block.
+getTipBlockHeader :: NetworkName -> ClientM (Hash "BlockHeader", BlockHeader)
```

Comment:
Here, I'd recommend to have a client in the form of:

```hs
data Client m = Client m
    { getBlockByHash :: NetworkName -> Hash "BlockHeader" -> m Block
    ...
    } 
```

with something like:

```hs
createClient :: Manager -> BaseUrl -> Client m
```

This to prevent the Servant types to spread too far in the code. It also makes things much easier for the consumer of that module API to use those functions. How they are implemented isn't relevant.

Reply:
We have put that data type in #41.

Reply:
Resolving, as we will add this in a further PR.